### PR TITLE
DH-664 save event acceptance tests

### PR DESCRIPTION
**Pull request by yeahfro**
This change hooks up the acceptance tests for saving an event. Included:
- General tidy up of scenarios and steps
- Save event acceptance tests

Still to come:
- Refactor code further

screen shot 2017-09-22 at 14 08 13

### CODE REVIEW

File: test/acceptance/features/events/save-new-event.feature [hunk 1]
```diff
@@ -7,29 +7,24 @@ Feature: Save a new Event in Data hub
   Background:
     Given I am an authenticated user on the data hub website
 
-  @save-new-event-submit @ignore
+  @save-new-event-submit
```
[the remark this answers is not in the document]
Comment (feedmypixel) [reply test/acceptance/features/events/save-new-event.feature]: @tyom also has some thoughts around a `BEM` style syntax for this.

File: test/acceptance/features/events/step_definitions/event-create.js [hunk 2]
```diff
@@ -152,17 +150,31 @@ defineSupportCode(({ Given, Then, When }) => {
...
 
   When(/^I add it to the programmes list$/, async () => {
     await Events
       .click('@addAnotherProgramme')
   })
 
+  When(/^I enter all mandatory fields related to the event$/, async () => {
+    await Events
+      .setValue('@eventName', faker.company.companyName())
+      .selectListOption('event_type', 2)
```
Comment (feedmypixel) [inline test/acceptance/features/events/step_definitions/event-create.js]: There is a custom command `getListOption.js` helper in #666 that randomly chooses an option from a multiple select. May help here.

Comment (yeahfro) [reply test/acceptance/features/events/step_definitions/event-create.js]: I will refactor once #666 goes in.

File: test/acceptance/features/events/step_definitions/event-create.js [hunk 3]
```diff
@@ -126,7 +124,7 @@ defineSupportCode(({ Given, Then, When }) => {
   When(/^I select shared team ([0-9])$/, async (optionNumber) => {
     await Events
       .setValue('@sharedTeams', '')
-      .selectTeam(optionNumber)
+      .selectListOption('teams', optionNumber)
```
[the remark this answers is not in the document]
Comment (yeahfro) [reply test/acceptance/features/events/step_definitions/event-create.js]: I will refactor once #666 goes in.

File: test/acceptance/features/events/save-new-event.feature [hunk 4]
```diff
@@ -7,29 +7,24 @@ Feature: Save a new Event in Data hub
...
   Scenario: Verify event is submitted
 
-    When I navigate to create an event page
+    When I navigate to the create an event page
     And I enter all mandatory fields related to the event
-    And I click on save button
-    Then I see the Added new event confirmation message
+    And I click the save button
+    Then I see the "Event created" success message
```
[the remark this answers is not in the document]
Comment (yeahfro) [reply test/acceptance/features/events/save-new-event.feature]: I will refactor once #667 goes in.

File: test/acceptance/features/events/save-new-event.feature [hunk 5]
```diff
@@ -25,6 +25,6 @@ Feature: Save a new Event in Data hub
     And I verify the event address line 1 has an error message
     And I verify the event address town has an error message
     And I verify the event address country has an error message
-    Then I see the "Event created" success message is not displayed
+    Then I see the success message
```
Comment (venureddy95) [inline test/acceptance/features/events/save-new-event.feature]: @yeahfro : Are we checking for success message here?

**Pull request merged**